imageloading-core/src/main/java/com/google/accompanist/imageloading/AsyncImage.kt

```diff
@@ -0,0 +1,367 @@
+ *
+ * @param state The request to execute.
+ * @param modifier [Modifier] used to adjust the layout algorithm or draw decoration content.
+ * @param alignment Optional alignment parameter used to place the loaded [ImageBitmap] in the
+ * given bounds defined by the width and height.
+ * @param contentScale Optional scale parameter used to determine the aspect ratio scaling to be
+ * used if the bounds are a different size from the intrinsic size of the loaded [ImageBitmap].
+ * @param colorFilter Optional colorFilter to apply for the [Painter] when it is rendered onscreen.
+ * @param fadeIn Whether to run a fade-in animation when images are successfully loaded.
+ * Default: `false`.
+ * @param fadeInDurationMs Duration for the fade animation in milliseconds when [fadeIn] is enabled.
+ * @param previewPlaceholder Drawable resource ID which will be displayed when this function is
+ * ran in preview mode.
+ * @param shouldRefetchOnSizeChange Lambda which will be invoked when the size changes, allowing
+ * optional re-fetching of the image. Return true to re-fetch the image.
+ */
+@Composable
+fun <R : Any> AsyncImage(
+    state: AsyncImageState<R>,
+    contentDescription: String?,
+    modifier: Modifier = Modifier,
+    alignment: Alignment = Alignment.Center,
+    contentScale: ContentScale = ContentScale.Fit,
+    colorFilter: ColorFilter? = null,
+    fadeIn: Boolean = false,
+    fadeInDurationMs: Int = DefaultTransitionDuration,
+    @DrawableRes previewPlaceholder: Int = 0,
+    shouldRefetchOnSizeChange: (currentResult: ImageLoadState, size: IntSize) -> Boolean = { _, _ -> false },
+) {
+    if (LocalInspectionMode.current && previewPlaceholder != 0) {
+        // If we're in inspection mode (preview) and we have a preview placeholder, just draw
+        // that using an Image and return
+        Image(
+            painter = painterResource(previewPlaceholder),
+            contentDescription = null,
+            modifier = modifier,
+        )
+        return
+    }
+
+    val coroutineScope = rememberCoroutineScope()
+
+    val imageLoader = remember(state, coroutineScope) {
+        ImageLoader(
+            state = state,
+            coroutineScope = coroutineScope,
+            shouldRefetchOnSizeChange = shouldRefetchOnSizeChange,
+        )
+    }
+
+    val cf = if (fadeIn && state.loadState is ImageLoadState.Success) {
+        val fadeInTransition = updateFadeInTransition(
+            key = imageLoader.state,
+            durationMs = fadeInDurationMs
+        )
+        remember { ColorMatrix() }
+            .apply {
+                updateAlpha(fadeInTransition.alpha)
+                updateBrightness(fadeInTransition.brightness)
+                updateSaturation(fadeInTransition.saturation)
+            }
+            .let { matrix ->
+                ColorFilter.colorMatrix(matrix)
+            }
+    } else {
+        // If fade in isn't enabled, just use the provided `colorFilter`
```

Comment:
hm, don't you think it is strange to ignore the passed colorFilter while we animating. so the animation is not animating to the final color filter in this case. we can't combine two color filteres I guess?